network/router.go

```diff
@@ -40,6 +40,10 @@ type Router struct {
 
 	// wg waits for all handleConn routines to be done.
 	wg sync.WaitGroup
+
+	//if not nil, is called by this entity when any network error occurs (e.g. TCP stream broken, host disconnected, etc)
+	//should be set by using SetErrorHandler(). The 1st argument is the remote server with whom the error happened
+	errorHappened func(*ServerIdentity)
```

Comment:
`connDropped` ? `errorHappened` is like `whatever thing is wrong I put it here`. It does not *mean* much. Feel free to propose other names.